docs/developer/best-practices/security.asciidoc

```diff
@@ -1,55 +1,129 @@
+pollution attack in NodeJS application | Olivier Arteau]
+
+=== Server-Side Request Forgery (SSRF) ===
+
+https://owasp.org/www-community/attacks/Server_Side_Request_Forgery[_OWASP reference for SSRF_]
+
+SSRF is a class of attacks where a vulnerable server is forced to make an unintended request, usually to an HTTP API. This is often used as
+a vector for information disclosure or injection attacks.
+
+*Best practices:*
+
+* Ensure that all outbound requests from the {kib} server use hard-coded URLs if possible.
+* If user input is used to construct a URL for an outbound request, ensure that an allow-list is used to validate the endpoints and that user input is escaped properly. Ideally
+this allow-list should be set in `kibana.yml` so only server administrators can change it.
+** This is particularly relevant when using `transport.request` with the {es} client, as no automatic escaping is performed.
+
+=== Reverse Tabnabbing ===
+
+https://owasp.org/www-community/attacks/Reverse_Tabnabbing[_OWASP reference for Reverse Tabnabbing_]
+
+Reverse Tabnabbing is an attack where a link to a malicious page is used to rewrite a vulnerable parent page. This is often used as a vector
+for phishing attacks. {kib} defends against this primarily by widespread use of the EUI framework, which automatically adds the `rel`
+attribute to anchor tags, buttons, and other vulnerable DOM elements.
+
+*Best practices:*
+
+* Use EUI components to build your UI whenever possible. Otherwise, ensure that any DOM elements that have an `href` attribute also have the
+`rel="noreferrer noopener"` attribute specified. For more information, see the
+https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/HTML5_Security_Cheat_Sheet.md#tabnabbing[OWASP HTML5 Security Cheat
+Sheet].
+* Allowing users to input markdown is a common culprit. If using a non-EUI markdown renderer, use a custom link renderer for rendered links.
+
+=== Information Disclosure ===
+
+Information Disclosure is not an attack, but it describes whenever sensitive information is accidentally revealed. This can be configuration
+info, stack traces, or other data that the user has not been authorized to access. This is a widespread concern that cannot be addressed
+with a single security control, but at a high level {kib} relies on the hapi framework to automatically redact stack traces and detailed
+error messages in HTTP 5xx response payloads.
+
+*Best practices:*
+
+* When writing your code, look for instances where sensitive information may be accidentally revealed, particularly in error messages, in
+the UI, and in URL parameters that are exposed to users.
+* Make sure that sensitive request data is not forwarded to external resources. For example, copying client request headers and using them
+to make an another request could accidentally expose the user's session cookie.
```

Comment:
__note:__ `could accidentally expose the user's session cookie.` - actually the current state of things is even worse - after successful authentication request is mutated with `Authorization: decrypted-credentials` header (for BWC reasons) and hence this header is accessible from any request handler that requires authentication. That's something we want to stop doing in the future though.

Reply:
I'll change this to a broader statement, "could acccidentally expose the user's credentials"